Composer/packages/client/src/pages/design/DebugPanel/TabExtensions/WebchatLog/WebChatInspectorPane.tsx

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import { JsonEditor } from '@bfc/code-editor';
+import { css, jsx } from '@emotion/core';
+import { NeutralColors } from '@uifabric/fluent-theme';
+import { Pivot, PivotItem } from 'office-ui-fabric-react/lib/Pivot';
+import { useCallback } from 'react';
+import { Resizable } from 're-resizable';
+
+import { WebChatInspectionData } from '../../../../../recoilModel/types';
+
+// TODO: make this complete
+type ActivityType = 'conversationUpdate' | 'message' | 'trace';
+
+const editorStyles = css`
+  border: none;
+`;
+
+const pivotStyles = css`
+  margin-bottom: 8px;
+`;
+
+type WebChatInspectorPaneProps = {
+  inspectionData?: WebChatInspectionData;
+  setInspectionData: (data: WebChatInspectionData) => void;
```

Comment:
onUpdateInspectionData, without ```on``` prefix it sounds like an imperative method. Also ```set``` prefix is usually used for state mutators

Reply:
I'm confused by this comment. This method is mutating state. Could you explain the scenarios in which you would use `onUpdateData` vs `setData` ?

Reply:
updates the state in the parent component, but for this component, it's a callback.
callbacks are named like OnSomethingHappened
state mutators are named like setSomething

Reply:
Got it, thanks :)